pytests/test_debugging.py

```diff
@@ -0,0 +1,504 @@
+    `script_hash` argument is omitted, the first parsed script is
+    going to be used.
+    """
+    def set_breakpoint(self, line_number: int, column_number: int, script_hash: str = None) -> Breakpoint:
+        script_hash = script_hash if script_hash is not None else self.scripts_parsed[0]["hash"]
+        message = DebuggerClientMessage.set_breakpoint(line_number, column_number, script_hash)
+        response = self.send_message(message)
+        self._env.assertContains("result", response[0])
+        self._env.assertContains("breakpointId", response[0]["result"])
+        # The resolved locations number must be greater than zero. If
+        # it is zero it indicates that the debugger couldn't resolve the
+        # breakpoint to any point in the script.
+        self._env.assertGreater(len(response[0]["result"]["locations"]), 0)
+        breakpoint = Breakpoint(self, response[0]["result"])
+        self.breakpoints.append(breakpoint)
+        return breakpoint
+
+    """
+    Removes the breakpoint with the id passed.
+    """
+    def remove_breakpoint(self, breakpoint_id: str):
+        response = self.send_message(DebuggerClientMessage.remove_breakpoint(breakpoint_id))
+        self._env.assertContains("result", response[0])
+        self._env.assertEqual({}, response[0]["result"])
+
+    """
+    Resumes the execution (if it was paused) and returns all the events
+    in response to the resume request.
+    """
+    def resume(self) -> [dict]:
+        self.pause = None
+        return self.send_message(DebuggerClientMessage.DEBUGGER_RESUME)
+
+    """
+    Resumes the execution (if it was paused), waits until the next
+    pause happens and returns all the messages in between.
+    """
+    def resume_and_wait_until_pauses(self) -> [dict]:
+        messages = self.resume()
+        while True:
+            message = self.receive_event()
+            messages.append(message)
+            if "method" in message and message["method"] == "Debugger.paused":
+                self.pause = message
+                break
+        return messages
+
+    """
+    Debugger: step into the stack frame.
+    """
+    def step_into(self) -> [dict]:
+        return self.send_message(DebuggerClientMessage.DEBUGGER_STEP_INTO)
+
+    """
+    Debugger: step out of the current stack frame.
+    """
+    def step_out(self) -> [dict]:
+        return self.send_message(DebuggerClientMessage.DEBUGGER_STEP_OUT)
+
+    """
+    Debugger: step over the next instruction.
+    """
+    def step_over(self) -> [dict]:
+        return self.send_message(DebuggerClientMessage.DEBUGGER_STEP_OVER)
+
+
+def deploy_script(env: Env):
+    env.expect('TFUNCTION', 'DELETE', 'foo')
+    env.expect('TFUNCTION', 'LOAD', 'debug', SCRIPT).contains('The V8 remote debugging server is waiting for a connection')
+
+@gearsTest()
+def testDeployConnectStartDisconnect(env: Env):
+    deploy_script(env)
+    client = DebuggerClient(env)
+    client.disconnect()
```

Comment:
Should we check something after? Can you remind me what is expected here?

Reply:
Okay, I am sending a PING and check that a reply is sent back. Is this sufficient? Do you suggest to check for anything else? We expect that the debugging session ends well: no crashes, no hangs, Redis can operate fine after.

Reply:
For example, do we expect the library to still remain on the server? If so lets check it and if not lets also check it?

Reply:
We can do that, okay.

Reply:
Checking now that the library is automatically deleted after we disconnect.